development/Dockerfile

```diff
@@ -1,49 +1,78 @@
-ARG NAUTOBOT_VER="1.2.11"
-ARG PYTHON_VER=3.8
-FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
+# -------------------------------------------------------------------------------------
+# Nautobot App Developement Dockerfile Template
+# Version: 1.1.0
+#
+# Apps that need to add additional steps or packages can do in the section below.
+# -------------------------------------------------------------------------------------
+# !!! USE CAUTION WHEN MODIFYING LINES BELOW
 
-# Make the value available after the FROM directive
-ARG NAUTOBOT_VER
+# Accepts a desired Nautobot version as build argument, default to 1.4
+ARG NAUTOBOT_VER="1.4"
 
-ENV prometheus_multiproc_dir=/prom_cache
+# Accepts a desired Python version as build argument, default to 3.8
+ARG PYTHON_VER="3.8"
 
+# Retrieve published development image of Nautobot base which should include most CI dependencies
+FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
+
+# Runtime argument and environment setup
 ARG NAUTOBOT_ROOT=/opt/nautobot
 
+ENV prometheus_multiproc_dir=/prom_cache
 ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
 
-RUN apt-get update && apt-get install -y gcc libmariadb-dev-compat libmariadb-dev
+# Install Poetry manually via its installer script;
+# We might be using an older version of Nautobot that includes an older version of Poetry
+# and CI and local development may have a newer version of Poetry
+# Since this is only used for development and we don't ship this container, pinning Poetry back is not expressly necessary
+# We also don't need virtual environments in container
+RUN curl -sSL https://install.python-poetry.org | python3 - && \
+  poetry config virtualenvs.create false
```

Comment:
What is the benefit of using the installer script versus just doing the install with pip?

Reply:
We are standardizing the install method and at present is the preferred way of installing Poetry.